Mathlib/Combinatorics/SimpleGraph/Sum.lean

```diff
@@ -0,0 +1,76 @@
+/-
+Copyright (c) 2022 Iván Renison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Iván Renison
+-/
+import Mathlib.Combinatorics.SimpleGraph.Basic
+import Mathlib.Combinatorics.SimpleGraph.Maps
+
+/-!
+# Disjoint sum of graphs
+
+This file defines the disjoint sum of graphs. The disjoint sum of `G : SimpleGraph α` and
+`H : SimpleGraph β` is a graph on `α ⊕ β` where `u` and `v` are adjacent if and only if they are
+both in `G` and adjacent in `G`, or they are both in `H` and adjacent in `H`.
+
+## Main declarations
+
+* `SimpleGraph.Sum`: The disjoint sum of graphs.
+
+## Notation
+
+* `G ⊎ H`: The disjoint sum of `G` and `H`.
+-/
+
+variable {α β γ : Type*}
+
+namespace SimpleGraph
+
+/-- Disjoint sum of `G` and `H`. -/
+@[simps!]
+protected def Sum (G : SimpleGraph α) (H : SimpleGraph β) : SimpleGraph (α ⊕ β) where
+  Adj u v := match u, v with
+    | Sum.inl u, Sum.inl v => G.Adj u v
+    | Sum.inr u, Sum.inr v => H.Adj u v
+    | _, _ => false
+  symm u v := match u, v with
+    | Sum.inl u, Sum.inl v => G.adj_symm
+    | Sum.inr u, Sum.inr v => H.adj_symm
+    | Sum.inl _, Sum.inr _ | Sum.inr _, Sum.inl _ => id
+  loopless u := by cases u <;> simp
+
+@[inherit_doc] infixl:60 " ⊎ " => SimpleGraph.Sum
```

Comment:
Should we use `⊕g` instead? @YaelDillies What do you think? Internet search says that `⊎` is called "multiset union" symbol and is used for `+` on `Multiset`s. Will it cause confusion?

Reply:
I prefer `⊕g` over `⊎` here too - it looks closer to the informal notation, and aligns with existing Lean notation for disjoint union.